src/babelRegister.js

```diff
@@ -20,15 +20,16 @@ function resolve(presetsOrPlugins, type) {
  * To prevent `babel-register` from reading foreign Babel
  * configs, `babelrc` has to be set to `false` and Haul's
  * Babel config inlined into `babel-register`'s options.
- * 
+ *
  * Also presets and plugins need to be resolved to
  * Haul's `node_modules` directory, otherwise Babel would
  * search from them in project's `node_modules`.
  */
 require('babel-register')(
   Object.assign(
     {
-      ignore: /node_modules(?!\/haul)/,
+      // Use a PNPM-compatible search pattern for node_modules.
```

Comment:
It occurred to me that this could just be done:

```js
const haulSrcPath = __dirname; // Should be the 'haul/src' folder.
ignore: function (filename) {
    return filename.slice(0, haulPath) !== haulPath); // Ignore modules which are not Haul src files.
}
```

My understanding is that the point of the regex was to scope Babel to process only the Haul package contents, regardless of whether Haul is run by itself or as a dependency. Using the function callback pattern would save on Regex compute time!